kernel: Added required dependencies for socket match.

oldium opened the pull request:
This applies to kernel 4.10 and newer.

See torvalds/linux@8db4c5b

The above commit, added to kernel 4.10, added new dependency
for building the NETFILTER_XT_MATCH_SOCKET (xt_socket.ko)
module. The NF_SOCKET_IPVx options (both of them) need to
be enabled in order to build the NETFILTER_XT_MATCH_SOCKET
module. Without the change the module is not built.

oldium committed:
kernel: Added required dependencies for socket match.

This applies to kernel 4.10 and newer.

See https://github.com/torvalds/linux/commit/8db4c5be88f62ffd7a552f70687a10c614dc697b

The above commit added to kernel 4.10 added new dependency
for building the NETFILTER_XT_MATCH_SOCKET (xt_socket.ko)
module. The NF_SOCKET_IPVx options (both of them) need to
be enabled in order to build the NETFILTER_XT_MATCH_SOCKET
module. Without the change the module is not built.

Signed-off-by: Oldřich Jedlička <oldium.pro@gmail.com>

dedeckeh commented:
Your PR has formal issues; the git commit description fails to explain why this change is required and the patch has no SoB. See https://openwrt.org/submitting-patches how patches need to be submitted.

oldium commented:
@dedeckeh Ok, sorry, forgot that. Updated description of pull request and of the commit to mention the reason of the change, added signed-off-by statement to commit.

hauke commented:
Thank you for the patch, it was applied to master.
